### PR TITLE
Update REB documentation

### PR DESCRIPTION
**Pull request by klaudiagrz**
**Description**

Changes proposed in this pull request:

- Update REB documentation with the info about namespaced resources
- Update diagrams

**Related issue(s)**
#297

### CODE REVIEW

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 1]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
 type: Architecture
 ---
 
-The Remote Environment Broker (REB) workflow starts with the registration process, during which a remote environment (RE) is registered on the Kyma cluster.
+The Remote Environment Broker (REB) workflow consists of the following steps:
 
-### Remote environment registration process
+1. The Remote Environment Broker watches for RemoteEnvironment (RE) custom resources in the cluster and EnvironmentMappings (EMs) in all Environments.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: Environments -> Namespaces

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 2]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+2. The user creates an EnvironmentMapping custom resource in a given Environment. The EnvironmentMapping has the same name as the RemoteEnvironment it represents.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: corresponds to

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 3]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+3. The Remote Environment Broker creates a `remote-env-broker` Service Broker (SB) inside a given Environment. This Service Broker contains data of all services provided by the RemoteEnvironments. There is always only one Service Broker per Environment, even if there are more EnvironmentMappings.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: environment in which EM is created

Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: provided by the RemoteEnvironments activated by EM

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 4]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+4. The Service Catalog fetches services that the `remote-env-broker` Service Broker (SB) exposes.
+5. The Service Catalog creates ServiceClasses inside a given Environment.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: creates ServiceClasses for each service received from the broker.

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 5]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
 type: Architecture
 ---
 
-The Remote Environment Broker (REB) workflow starts with the registration process, during which a remote environment (RE) is registered on the Kyma cluster.
+The Remote Environment Broker (REB) workflow consists of the following steps:
 
-### Remote environment registration process
+1. The Remote Environment Broker watches for RemoteEnvironments (REs) in the cluster and EnvironmentMappings (EMs) in all Namespaces.
+2. The user creates an EnvironmentMapping custom resource in a given Environment. The EnvironmentMapping activates the services offered by a given RemoteEnvironment and has the same name as the RemoteEnvironment it corresponds to.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: `The EnvironmentMapping activates the services offered by a given RemoteEnvironment and has the same name as the RemoteEnvironment it corresponds to.`

`The EnvironmentMapping activates the services offered by a given RemoteEnvironment which has the same name.`

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 6]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+3. The Remote Environment Broker creates a `remote-env-broker` Service Broker (SB) inside an Environment in which the EnvironmentMapping is created. This Service Broker contains data of all services provided by the RemoteEnvironments activated by EnvironmentMappings. There is always only one Service Broker per Environment, even if there are more EnvironmentMappings.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: `This Service Broker contains data of all services provided by the RemoteEnvironments activated by EnvironmentMappings.`

`This Service Broker contains data of all services provided by the activated RemoteEnvironments.`

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 7]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+4. The Service Catalog fetches services that the `remote-env-broker` Service Broker exposes.
+5. The Service Catalog creates ServiceClasses for each service received from the Service Broker.
```
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: ServiceClasses -> ServiceClass

File: docs/service-brokers/docs/020-architecture-reb.md (hunk 5 above)
Comment (polskikiel) [inline docs/service-brokers/docs/020-architecture-reb.md]: `RemoteEnvironments (REs) EnvironmentMappings (EMs)`
Why do you introduce the abbreviations if you are not using them?

Comment (klaudiagrz) [reply docs/service-brokers/docs/020-architecture-reb.md]: They are used in the diagram.

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 8]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
 type: Architecture
 ---
 
-The Remote Environment Broker (REB) workflow starts with the registration process, during which a remote environment (RE) is registered on the Kyma cluster.
+The Remote Environment Broker (REB) workflow consists of the following steps:
 
-### Remote environment registration process
+1. The Remote Environment Broker watches for RemoteEnvironments (REs) in the cluster and EnvironmentMappings (EMs) in all Namespaces.
```
Comment (kazydek) [inline docs/service-brokers/docs/020-architecture-reb.md]: and not all Environments?

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 9]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+2. The user creates an EnvironmentMapping custom resource in a given Environment. The EnvironmentMapping activates the services offered by a RemoteEnvironment of the same name.
+3. The Remote Environment Broker creates a `remote-env-broker` Service Broker (SB) inside an Environment in which the EnvironmentMapping is created. This Service Broker contains data of all services provided by the activated RemoteEnvironments. There is always only one `remote-env-broker` Service Broker per Environment, even if there are more EnvironmentMappings.
+4. The Service Catalog fetches services that the `remote-env-broker` Service Broker exposes.
+5. The Service Catalog creates ServiceClass for each service received from the Service Broker.
```
Comment (kazydek) [inline docs/service-brokers/docs/020-architecture-reb.md]: a ServiceClass ?

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 10]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
 type: Architecture
 ---
 
-The Remote Environment Broker (REB) workflow starts with the registration process, during which a remote environment (RE) is registered on the Kyma cluster.
+The Remote Environment Broker (REB) workflow consists of the following steps:
 
-### Remote environment registration process
+1. The Remote Environment Broker watches for RemoteEnvironments (REs) in the cluster and EnvironmentMappings (EMs) in all Namespaces.
+2. The user creates an EnvironmentMapping custom resource in a given Environment. The EnvironmentMapping activates the services offered by a RemoteEnvironment of the same name.
```
Comment (kazydek) [inline docs/service-brokers/docs/020-architecture-reb.md]: and not `the EnvironmentMapping custom resource` as it is the specific resource?

File: docs/service-brokers/docs/020-architecture-reb.md [hunk 11]
```diff
@@ -3,25 +3,17 @@ title: The Remote Environment Broker architecture
...
+3. The Remote Environment Broker creates a `remote-env-broker` Service Broker (SB) inside an Environment in which the EnvironmentMapping is created. This Service Broker contains data of all services provided by the activated RemoteEnvironments. There is always only one `remote-env-broker` Service Broker per Environment, even if there are more EnvironmentMappings.
```
Comment (kazydek) [inline docs/service-brokers/docs/020-architecture-reb.md]: and not `inside the Environment`? -> the one in which the EnvironmentMapping is created

Comment (kazydek) [inline docs/service-brokers/docs/020-architecture-reb.md]: and not `inside the Environment`? the specific one, the one in which the EnvironmentMapping is created

File: docs/service-brokers/docs/020-architecture-reb.md (hunk 8 above)
Comment (klaudiagrz) [reply docs/service-brokers/docs/020-architecture-reb.md]: Actually, originally I've also written `all Environments` but during the technical review, I was asked to change Environments into Namespaces.

File: docs/service-brokers/docs/020-architecture-reb.md (hunk 11 above)
Comment (klaudiagrz) [reply docs/service-brokers/docs/020-architecture-reb.md]: Good point, thank you!

File: docs/service-brokers/docs/020-architecture-reb.md (hunk 8 above)
Comment (klaudiagrz) [reply docs/service-brokers/docs/020-architecture-reb.md]: @polskikiel Could you explain why we should use Namespace instead of Environment here?

Comment (polskikiel) [reply docs/service-brokers/docs/020-architecture-reb.md]: @klaudiagrz @kazydek From the technical point of view REB watches for EnvironmentMappings in all Namespaces. Not only in the Environments.

**Comment by klaudiagrz** [comment]
@PK85 Could you have a look and leave your feedback?

File: docs/service-brokers/docs/020-architecture-reb.md (hunk 10 above)
Comment (PK85) [inline docs/service-brokers/docs/020-architecture-reb.md]: "of the same name" sounds like a magic, please clearly say that EM must has same name as RE that will be activated in a given Environment

**Pull request merged**